### PR TITLE
Discovery separate point box view

### PR DESCRIPTION
**Pull request by xuezhaokun**
Separate point view and box view for discovery map interface.
@mjstealey Can you help me deploy it to some testing environment to let people try it out? Thanks!

### CODE REVIEW

**Comment by mauriel133** [comment]
@xuezhaokun, I'm trying to deploy this so I can test it at http://park.hydroshare.org/ but I've had no luck so far. I'm not sure why the deployment is failing.

**Comment by xuezhaokun** [comment]
@mauriel133 Is there any specific error reported?

**Comment by mauriel133** [comment]
@xuezhaokun, no. It just stops and error is thrown.

**Comment by xuezhaokun** [comment]
@mauriel133 I just did a rebuild on my local VM. It works fine. I am not sure what is the problem...

**Comment by mauriel133** [comment]
Zhaokun,

@xuezhaokun,

I'm preparing to test this PR, but before I do so we need to work out the recent merge conflicts with the Inline Event Handlers pull request. 

I took a stab at solving these merge conflicts. Attached is the file that I ended up with. Please test it and let me know if everything works as expected after the merge and I will then push this change.

[discover.zip](https://github.com/hydroshare/hydroshare/files/923935/discover.zip)

**Comment by xuezhaokun** [comment]
@martinseul 
I just tested the file you provided. There are couple issues.

1. The clusters cannot be shown, and the images in legend are not loaded.
image

2. When I tried to do faceting on point view, it does not work now. 
image

3.The map click function does not work consistently with the table. Now, when I click on a marker, the corresponding resource in the table is not selected.
image

**Comment by pkdash** [comment]
@xuezhaokun What's the status of this PR?

**Comment by xuezhaokun** [comment]
@pkdash Something does not work well after moving out the js file. I need to debug this PR.

**Comment by alvacouch** [comment]
@xuezhaokun We've reviewed this and have one favor to ask. **Could you turn off clicking on areas in the map?** Leave clicking on clusters and markers alone. Users find clicking on areas too confusing. 

I hope this would take a few seconds. 

Best wishes and have fun with your internship!

**Comment by xuezhaokun** [comment]
@alvacouch OK. So now, we just  highlight a box resource, only when users click on the resource's center marker, right? I think I just need to remove all code related to "area click" then. I do not have my laptop with me now. I finish this later this evening.
Thanks!

**Comment by alvacouch** [comment]
@xuezhaokun Yes, that's correct. The area clicks are not popular. The marker clicks do what people want. Many thanks!

**Comment by xuezhaokun** [comment]
@alvacouch It should be ready for testing.

**Comment by alvacouch** [comment]
@Maurier @aphelionz I had to rebase against develop. The only real changes are to discovery. I hope this can be reviewed soon, because we need it for AGU.

**Comment by Maurier** [comment]
@xuezhaokun 
While testing this I see that clicking on a box coverage center highlights an area outside of it. Is this a correct behavior?
image

**Comment by xuezhaokun** [comment]
@Maurier I think it is Google Map's issue. For this data source, it is supposed to draw a rectangle like this
image
But Google Map only draws the red rectangle, the one with coordinates on the east part of the earth.I will a look at this issue.

**Comment by xuezhaokun** [comment]
@Maurier should be fixed now.

**Comment by Maurier** [comment]
@xuezhaokun could you update the deployment at http://cuahsi-dev-1.hydroshare.org with your most recent change?

**Comment by hydrobuild** [comment]
Build finished.

File: config/hydroshare-config.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ### Local Configuration Variables ###
 HS_PATH: /home/hydro/hydroshare
 HOST_SSL_DIR: /opt/hs-certs
-HS_DATABASE: pg.hydroshare-10-10-17.sql
+HS_DATABASE: pg.hydroshare-10-29-17.deploy.sql
```
Comment (aphelionz) [inline config/hydroshare-config.yaml]: We should not be committing changes to this file

File: hydroshare/local_settings.py [hunk 2]
```diff
@@ -107,18 +107,18 @@
...
+USE_IRODS=True
+IRODS_ROOT='/tmp'
+IRODS_ICOMMANDS_PATH='/usr/bin'
+IRODS_HOST='cuahsi-irods-1.hydroshare.org'
+IRODS_PORT='1247'
+IRODS_DEFAULT_RESOURCE='cuahsi2ReplResc'
+IRODS_HOME_COLLECTION='/hydroshareZone/home/cuahsi2DataProxy'
+IRODS_CWD='/hydroshareZone/home/cuahsi2DataProxy'
+IRODS_ZONE='hydroshareZone'
+IRODS_USERNAME='cuahsi2DataProxy'
```
Comment (aphelionz) [inline hydroshare/local_settings.py]: Same here. I don't believe we want these changes merged into develop

File: theme/static/js/discover.js [hunk 3]
```diff
@@ -952,12 +952,18 @@ var updateListView = function (data) {
...
+    var djangoTemplateStart = new Date();
     $.ajax({
         type: "GET",
         url: requestURL,
         data: data,
         dataType: 'html',
         success: function (data) {
+            var djangoTemplateEnd = new Date();
+            var djangoElapsed = djangoTemplateEnd - djangoTemplateStart;
+            console.log("Django template elapsed: " + djangoElapsed);
```
Comment (aphelionz) [inline theme/static/js/discover.js]: We probably don't want benchmarking code in dev or prod

File: theme/static/js/discover.js [hunk 4]
```diff
@@ -1348,6 +1385,9 @@ $(document).ready(function () {
                         initMap(json_point_results, "point_map");
                         setMapItemsList(json_point_results, "point_map", null);
                         $("#discover-point-map-loading-spinner").hide();
+                        var pointMapInitRenderEnd = new Date();
+                        var pointMapInitRenderElapsed = pointMapInitRenderEnd - pointMapInitRenderStart;
+                        console.log("point map init elapsed: " + pointMapInitRenderElapsed);
```
Comment (aphelionz) [inline theme/static/js/discover.js]: same as above and for all similar examples in this file - we shouldn't be committing benchmarking code

**Comment by alvacouch** [comment]
Different PRs forthcoming.

**Pull request closed without merging**